src/Processors/QueryPlan/WindowStep.cpp

```diff
@@ -60,6 +62,8 @@ WindowStep::WindowStep(
 
 void WindowStep::transformPipeline(QueryPipelineBuilder & pipeline, const BuildQueryPipelineSettings &)
 {
+    auto num_streams = pipeline.getNumThreads();
```

Comment:
maybe `getNumStreams()`?

Reply:
`getNumStreams()` return 1 if there is a `ORDER BY` in the window,

Reply:
so, we don't _preserve_ the number of stream, but rather always fan out after the `WindowStep`

Reply:
exactly.

Reply:
then I think we should name the setting accordingly